docs/sequence/error_scenarios/PISPLinkingGrantConsentPhase2.puml

```diff
@@ -0,0 +1,256 @@
+  scopes: model.data.scopes
+}
+end note
+Switch --> DFSP_LM: 202 ACCEPTED
+deactivate DFSP_LM
+Switch ->  PISP_TP_IN: POST /consents
+activate PISP_TP_IN
+PISP_TP_IN -> PISP_TP_IN: const model = await loadFromKVS({key: 6789})
+PISP_TP_IN -> PISP_LM: model.registerCredential()
+activate PISP_LM
+PISP_TP_IN --> Switch: 202 Accepted
+deactivate PISP_TP_IN
+deactivate Switch
+rnote right of PISP_LM: state: consentRecieved
+rnote right of PISP_LM #LightBlue
+It's important to save the model with the consentId from this point onwards!
+Requests will not contain a consentRequestId in the upcoming requests.
+This may require changing `PersistentModel` to have a secondary key it can
+store the model with.
+end note
+PISP_LM --> PISP_LM: const challenge = deriveChallenge(consentRequest)
+PISP_LM --> PISP_LM: this.saveToKVS({key: '1a2b3c4d'})
+
+PISP_LM -> PISP_LM: ThirdpartyRequests.putConsents()
+rnote right of PISP_LM: state: signedConsentSent
+PISP_LM -> Switch: PUT /consents/1a2b3c4d
+activate Switch
+
+rnote right of PISP_LM #LightBlue
+PUT /consents/1a2b3c4d
+FSIOP-Source: pispa
+FSIOP-Destination: dfspa
+{
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }],
+  credential: {
+    credentialType: "FIDO",
+    status: "PENDING",
+    payload: PublicKeyCredential
+  }
+}
+end note
+
+Switch --> PISP_LM: 202 Accepted
+deactivate PISP_LM
+Switch -> DFSP_TP_IN: PUT /consents/1a2b3c4d
+activate DFSP_TP_IN
+
+DFSP_TP_IN -> DFSP_TP_IN: const model = await loadFromKVS({key: 1a2b3c4d})
+rnote right of DFSP_TP_IN: state: signedConsentRecieved
+DFSP_TP_IN -> DFSP_LM: model.validateSignedConsent()
+activate DFSP_LM
+rnote right of DFSP_TP_IN: state: pendingRegistrationAndValidation
+DFSP_TP_IN --> Switch: 202 Accepted
+deactivate Switch
+deactivate DFSP_TP_IN
+DFSP_LM -> DFSP_LM: Check signed consent.
+rnote right of DFSP_LM #LightBlue
+Does this need a backend request or can the scheme adapter check the consent.
+end note
+rnote right of DFSP_LM: state: signedConsentChecked
+DFSP_LM-> DFSP_LM: model.validateWithAuthService()
+DFSP_LM -> Switch: POST /consents/1a2b3c4d
+activate Switch
+
+rnote left of DFSP_LM #LightBlue
+POST /consents/1a2b3c4d
+FSIOP-Source: dfspa
+FSIOP-Destination: central-auth
+{
+  scopes: [{
+    accountId: 'dfspa.username.1234',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+    accountId: 'dfspa.username.5678',
+    actions: ['accounts.getBalance', 'accounts.transfer'],
+  }],
+  credential: {
+    credentialType: "FIDO",
+    status: "PENDING",
+    payload: PublicKeyCredential
+  }
+}
+end note
+Switch --> DFSP_LM: 202 Accepted
+DFSP_LM -> DFSP_LM: this.saveToKVS()
+deactivate DFSP_LM
+
+Switch -> AUTH: POST /consents/1a2b3c4d
+activate AUTH
+AUTH --> Switch: 202 Accepted
+AUTH -> AUTH: Check consent.
+AUTH -> Switch: PUT /consents/1a2b3c4d/error
```

Comment:
Please explain why this errored. Perhaps the public key of the credential mismatches the signed challenge?

Reply:
Auth Service determines consent is invalid. As to what triggers the auth-service to find it invalid I'm not too sure on the implementation so it's left very generalized.

Reply:
I think we just want an example in the sequence diagram so I can understand easily.